packages/orbit-components/src/primitives/TooltipPrimitive/components/TooltipContent.js

```diff
@@ -113,69 +114,91 @@ const TooltipContent = ({
-    () => sortPositionsAndAligns(preferredPosition, preferredAlign, theme),
-    [preferredAlign, preferredPosition, theme],
-  );
-
-  const dimensions = useDimensions({ containerRef, tooltip, content }, children, parent);
-  const position = React.useMemo(() => calculateTooltipPosition(positions, dimensions), [
-    dimensions,
-    positions,
-  ]);
-
-  const align = React.useMemo(() => calculateTooltipAlign(position, aligns, dimensions), [
-    aligns,
-    dimensions,
-    position,
-  ]);
+  const [tooltip, setTooltipRef] = React.useState(null);
+  const [arrowRef, setArrowRef] = React.useState(null);
+  const [contentHeight, setContentHeight] = React.useState(0);
+  const content = React.useRef<HTMLDivElement | null>(null);
+
+  const { styles, attributes: attrs, update } = usePopper(referenceElement, tooltip, {
+    placement,
+    modifiers: [
+      {
+        name: "arrow",
+        options: {
+          element: arrowRef,
+        },
+      },
+      {
+        name: "offset",
+        options: {
+          offset,
+        },
+      },
+      {
+        name: "flip",
+        enabled: !noFlip,
+      },
+    ],
+  });
+
+  React.useEffect(() => {
+    if (update) update();
+    if (content.current) setContentHeight(content.current.clientHeight);
+  }, [update, setContentHeight]);
+
+  const { popper, arrow } = styles;
 
   const handleInnerClick = React.useCallback(
     ev => {
-      if (tooltip.current) {
-        const focusableElements = tooltip.current.querySelectorAll(FOCUSABLE_ELEMENT_SELECTORS);
+      if (tooltip) {
+        const focusableElements = tooltip.querySelectorAll(FOCUSABLE_ELEMENT_SELECTORS);
         if (Object.values(focusableElements).some(v => v === ev.target)) {
           onClose();
           onCloseMobile();
         }
       }
     },
-    [onClose, onCloseMobile],
+    [onClose, onCloseMobile, tooltip],
   );
 
   return (
     <StyledTooltip role="tooltip" id={tooltipId} data-test={dataTest}>
       <StyledTooltipWrapper
-        shown={shown && position && align}
+        shown={shown}
         size={size}
-        align={align}
-        position={position}
-        ref={tooltip}
+        ref={setTooltipRef}
         error={error}
         help={help}
-        containerTop={dimensions.containerTop}
-        containerLeft={dimensions.containerLeft}
-        containerHeight={dimensions.containerHeight}
-        containerWidth={dimensions.containerWidth}
-        tooltipHeight={dimensions.tooltipHeight}
-        tooltipWidth={dimensions.tooltipWidth}
-        contentHeight={dimensions.contentHeight}
         role="tooltip"
         aria-hidden={!shown}
         onMouseEnter={onEnter}
         onMouseLeave={onClose}
+        contentHeight={contentHeight}
         onClick={handleInnerClick}
+        top={popper.top}
+        left={popper.left}
+        right={popper.right}
+        bottom={popper.bottom}
+        position={popper.position}
+        transform={popper.transform}
```

Comment:
Would this be ok? Not sure if it would have some performance regressions. 🤔  Same with arrow styles.
```suggestion
        $popper={popper}
```

Reply:
not sure why with `$`, it can be just `popper={popper} than